app/helpers/settings_analysis_profile_helper.rb

```diff
@@ -0,0 +1,69 @@
+module SettingsAnalysisProfileHelper
+  private
+
+  def settings_analysis_profile_summary(selected_scan, categories, files, file_stats, registry, nteventlog)
+    summary = [settings_analysis_basic_info(selected_scan)]
+    unless selected_scan.mode == "Host"
+      summary.push(settings_analysis_catrgories(categories)) if categories
+    end
+    summary.push(settings_analysis_files(files, file_stats)) if files
+    unless selected_scan.mode == "Host"
+      summary.push(settings_analysis_registry_items(registry)) if registry
+    end
+    summary.push(settings_analysis_event_log_items(nteventlog)) if nteventlog
+    safe_join(summary)
+  end
+
+  def settings_analysis_basic_info(selected_scan)
+    data = {:title => _('Basic Information'), :mode => "settings_analysis_basic_info"}
+    data[:rows] = [
+      row_data(_('Name'), selected_scan.name),
+      row_data(_('Description'), selected_scan.description),
+      row_data(_('Type'), selected_scan.mode),
+      row_data(_('Type'), selected_scan.mode),
+    ]
+    miq_structured_list(data)
+  end
+
+  def settings_analysis_catrgories(categories)
+    rows = []
+    data = {:title => _('Categories'), :rows => [], :mode => "settings_analysis_catrgories"}
+    categories.sort.each do |category|
+      rows.push({:cells   => [{:value => category}]})
+    end
+    data[:rows] = rows
```

Comment:
This pattern can be more cleanly written  (prefer map over creating a temporary array)

```suggestion
    data = {:title => _('Categories'), :rows => [], :mode => "settings_analysis_catrgories"}
    data[:rows] = categories.sort.map do |category|
      {:cells   => [{:value => category}]}
    end
```

Reply:
Actually now that I wrote that you don't even need 2 lines:

```suggestion
    data = {
      :title => _('Categories'),
      :mode => "settings_analysis_catrgories",
      :rows => categories.sort.map do |category|
        {:cells => [{:value => category}]}
      end
    }
```

Reply:
This refactoring also applies to the other methods below, but not required for this PR - it can be done in a follow up if you want.

Reply:
had refactored the remaining code as suggested in this PR..